generic_m2m_field/models.py

```diff
@@ -84,6 +84,10 @@ def get_object_or_none(self, model_class, pk=None):
         related_object = get_object_or_none(qs)
         return related_object.object if related_object else None
 
+    def get_by_name(self, name):
+        related_object = self.filter(name=name).get()
```

Comment:
this will not work 

firstly
self.filter(name=name).get() == self.get(name=name)

but you didn't want to raise ObjectDoesNotExist exception. You want return None if object not exists

so self.filter(name=name).first() is right solution

Reply:
Omg... you are right... I wanted to throw `MultipleObjectsReturned` when there are more same related objects but I forgot the principal difference between `first` and `get`

Reply:
I think that name, content_type, and object id is constrant. So MultipleObjectsReturned cannot happen